### PR TITLE
Add metadata for the graph

### PR DESCRIPTION
**Pull request by iam-robi**
Chain interface is modify to support the graph metadata, which wil be used to generate subgraphs for each chain in the mono repo. 

Here, an example is given with main net information.

### CODE REVIEW

**Comment by iam-robi** [comment]
also the PR updates wagmi with the new rpc url provided by patrick ogrady

File: src/chains.ts [hunk 1]
```diff
@@ -43,6 +43,15 @@ export interface Chain {
     fee_collector?: string;
     staking?: StakingContract[];
   }
+  theGraph?: {
+    tokens_whitelist: string[];
```
Comment (HariSeldon23) [inline src/chains.ts:47]: @iam-robi from my understanding the new Subgraph dynamically picks up the whitelisted tokens. So is this then still necessary?

Also the Chain interface is getting pretty verbose. I think it makes more sense to have another interface for the Graph which then is included in the Chain interface

Comment (ClementMoyal) [reply src/chains.ts:47]: In the new script, the Subgraph's settings files are set up with the SDK https://github.com/wagmiog/exchange-contracts/blob/6b5cf3c5f5f84d4f4afa1a02aa7e4e68ff1c48c2/scripts/deploy-graph.js#L42
We  unfortunately can't get the token adresses' dynamically on each chain, and which tokens we want to be whitelisted on each chain. So we need to write the addresses on the SDK beforehand
The whitelist is only to calculate a token price.
Yes it makes more sense, i will do another interface for the Graph.

**Comment by HariSeldon23** [comment]
@SarjuHansaliya can you comment here on the Graph picking up values dynamically for the Pools? Just want to make sure we have full transparency here

**Comment by SarjuHansaliya** [comment]
> @SarjuHansaliya can you comment here on the Graph picking up values dynamically for the Pools? Just want to make sure we have full transparency here
> 
> 

Looks like, i am not seeing any file changes in this pr. Should i refer any other pr ?

**Pull request closed without merging**